Source/Core/Core/State.cpp

```diff
@@ -554,7 +629,99 @@ u64 GetUnixTimeOfSlot(int slot)
     return 0;
 
   constexpr u64 MS_PER_SEC = 1000;
-  return static_cast<u64>(header.time * MS_PER_SEC) + (DOUBLE_TIME_OFFSET * MS_PER_SEC);
+  return static_cast<u64>(header.legacy_header.time * MS_PER_SEC) +
+         (DOUBLE_TIME_OFFSET * MS_PER_SEC);
+}
+
+static void DecompressLZ4(std::vector<u8>& raw_buffer, u64 size, File::IOFile& f)
+{
+  raw_buffer.resize(size);
+
+  u64 total_bytes_read = 0;
+  while (true)
+  {
+    s32 compressed_data_len;
+    f.ReadArray(&compressed_data_len, 1);
+
+    auto compressed_data = std::make_unique<char[]>(compressed_data_len);
+    f.ReadBytes(compressed_data.get(), compressed_data_len);
+
+    // We need to specify the output buffer's size for safety. This may exceed the positive bound
+    // of int, causing the buffer size to be interpreted as a negative value.
+    u32 max_decompress_size = (u32)std::min((u64)LZ4_MAX_INPUT_SIZE, size - total_bytes_read);
+
+    int bytes_read =
+        LZ4_decompress_safe(compressed_data.get(), (char*)raw_buffer.data() + total_bytes_read,
+                            compressed_data_len, max_decompress_size);
+
+    if (bytes_read < 0)
+    {
+      PanicAlertFmtT("Internal LZ4 Error - decompression failed ({0}, {1}, {2}) \n"
+                     "Try loading the state again",
```

Comment:
I see that this line was here before but I'd argue it makes no sense? Why would loading the state a second time change anything?

Reply:
I agree. I should've taken this out!